components/script/dom/htmloptionscollection.rs

```diff
@@ -0,0 +1,194 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use dom::bindings::codegen::Bindings::ElementBinding::ElementMethods;
+use dom::bindings::codegen::Bindings::HTMLCollectionBinding::HTMLCollectionMethods;
+use dom::bindings::codegen::Bindings::HTMLOptionsCollectionBinding;
+use dom::bindings::codegen::Bindings::HTMLOptionsCollectionBinding::HTMLOptionsCollectionMethods;
+use dom::bindings::codegen::Bindings::NodeBinding::NodeBinding::NodeMethods;
+use dom::bindings::codegen::UnionTypes::{HTMLOptionElementOrHTMLOptGroupElement, HTMLElementOrLong};
+use dom::bindings::error::{Error, ErrorResult};
+use dom::bindings::global::GlobalRef;
+use dom::bindings::inheritance::Castable;
+use dom::bindings::js::{JS, MutHeap, Root, RootedReference};
+use dom::bindings::reflector::reflect_dom_object;
+use dom::bindings::str::DOMString;
+use dom::element::{Element, ElementCreator};
+use dom::htmlcollection::{CollectionFilter, HTMLCollection};
+use dom::htmloptionelement::HTMLOptionElement;
+use dom::node::{document_from_node, Node};
+use dom::window::Window;
+use string_cache::{Atom, QualName};
+
+#[dom_struct]
+pub struct HTMLOptionsCollection {
+    collection: HTMLCollection,
+    root: MutHeap<JS<Node>>
+}
+
+impl HTMLOptionsCollection {
+    fn new_inherited(root: &Node, filter: Box<CollectionFilter + 'static>) -> HTMLOptionsCollection {
+        HTMLOptionsCollection {
+            collection: HTMLCollection::new_inherited(root, filter),
+            root: MutHeap::new(root)
+        }
+    }
+
+    pub fn new(window: &Window, root: &Node, filter: Box<CollectionFilter + 'static>)
+        -> Root<HTMLOptionsCollection>
+    {
+        reflect_dom_object(box HTMLOptionsCollection::new_inherited(root, filter),
+                           GlobalRef::Window(window),
+                           HTMLOptionsCollectionBinding::Wrap)
+    }
+
+    fn add_new_elements(&self, count: u32) -> ErrorResult {
+        let root = self.root.get();
+        let document = document_from_node(root.r());
+        let name_atom = Atom::from(DOMString::from("option".to_owned()));
+
+        for _ in 0..count {
+            let name = QualName::new(ns!(html), name_atom.clone());
+            let element = Element::create(name, None, document.r(), ElementCreator::ScriptCreated);
+            let node = element.r().upcast::<Node>();
+            try!(Node::pre_insert(node, root.r(), None));
```

Comment:
This looks pretty messy, you can achieve the same simply by using the AppendChild WebIDL function on Node and `HTMLOptionElement::new`.